engine/include/Engine/Intermediate/GameObject.inl

```diff
@@ -0,0 +1,179 @@
+#include "Engine/Intermediate/GameObject.hpp"
+
+GameObject::GameObject(const CreateArg& arg)
+    : m_name{arg.name}, m_pTransform{&DataChunk<TransformComponent>::getInstance()->addComponent(*this,
+                                                                                                 arg.transformArg)},
+      m_pComponents{}
+{
+}
+
+GameObject::GameObject()
+    : m_name{""}, m_pTransform{&DataChunk<TransformComponent>::getInstance()->addComponent(*this)}, m_pComponents{}
+{
+}
+
+template <typename T>
+void GameObject::updateComponentLink(const T* oldPtr, T* newPtr) noexcept
+{
+    GPE_ASSERT(oldPtr != nullptr, "oldPtr cannot be nullptr");
+    GPE_ASSERT(newPtr != nullptr, "newPtr cannot be nullptr");
+
+    for (Component* pComponent : m_pComponents)
+    {
+        T* checkedCompPtr = dynamic_cast<T*>(pComponent);
```

Comment:
I don't understand the use of dynamic_cast here.
if oldPtr is not nullptr, if checkedCompPtr and oldPtr points towards the same object, wouldn't the class be already the same ?